lib/routes/charge/items/catalog_item.dart

```diff
@@ -0,0 +1,108 @@
+      this._lastItem, this._addItem);
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(alignment: Alignment.bottomCenter, children: <Widget>[
+      Slidable(
+        actionPane: SlidableDrawerActionPane(),
+        actionExtentRatio: 0.25,
+        secondaryActions: <Widget>[
+          IconSlideAction(
+            caption: 'Delete',
+            color: Colors.red,
+            icon: Icons.delete_forever,
+            onTap: () {
+              DeleteItem deleteItem = DeleteItem(_itemInfo.id);
+              posCatalogBloc.actionsSink.add(deleteItem);
+              deleteItem.future.then((_) => showFlushbar(context,
+                  message: "${_itemInfo.name} is successfully deleted"));
+            },
+          ),
+        ],
+        key: Key(_itemInfo.id.toString()),
+        child: ListTile(
+          leading: _buildCatalogItemAvatar(),
+          title: Text(
+            _itemInfo.name,
+            style: theme.transactionTitleStyle,
+            overflow: TextOverflow.ellipsis,
+          ),
+          trailing: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.end,
+            children: <Widget>[
+              Row(
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
+                  children: <Widget>[
+                    Text(
+                      _formattedPrice(),
+                      style: theme.transactionAmountStyle,
+                    )
+                  ]),
+            ],
+          ),
+          onTap: () {
+            _addItem(accountModel, _itemInfo.currency, _itemPriceInSat());
+          },
+        ),
+      ),
+      Divider(
+        height: 0.0,
+        color: _lastItem
+            ? Color.fromRGBO(255, 255, 255, 0.0)
+            : Color.fromRGBO(255, 255, 255, 0.12),
+        indent: 72.0,
+      ),
+    ]);
+  }
+
+  String _formattedPrice({bool userInput = false, bool includeSymbol = true}) {
+    return Currency.fromSymbol(_itemInfo.currency) != null
+        ? Currency.fromSymbol(_itemInfo.currency).format(
+            Int64(_itemInfo.price.toInt()),
+            userInput: userInput,
+            includeSymbol: includeSymbol)
+        : accountModel.fiatConversionList
+            .firstWhere((f) => f.currencyData.shortName == _itemInfo.currency)
+            .format(accountModel.fiatConversionList
+                .firstWhere(
+                    (f) => f.currencyData.shortName == _itemInfo.currency)
+                .fiatToSat(_itemInfo.price));
+  }
+
+  Int64 _itemPriceInSat() {
+    return Currency.fromSymbol(_itemInfo.currency) != null
+        ? Currency.fromSymbol(_itemInfo.currency)
+            .parse(_formattedPrice(userInput: true, includeSymbol: false))
+        : accountModel.fiatConversionList
+            .firstWhere((f) => f.currencyData.shortName == _itemInfo.currency)
```

Comment:
For the code to be simpler I think we can add a function to the account model that return a fiat currency by `shortName`.

Reply:
[Done](https://github.com/breez/breezmobile/pull/289/commits/ea81c58a4279e8541d8c02d1f167e6562a9d9b9d)